### ISSUE TITLE
Post Link Preview Doesn't Conform to Permalink Settings

### ISSUE THREAD

**Issue by davidlim** [WordPress/gutenberg#65412]
## What problem does this address?

WordPress Core 6.6 updated the link preview on the sidebar of the editor to show the slug. For sites that use a custom permalink structure, the link shown on the sidebar is not the complete link unless you click on the link.

![unconforming-post-title](https://github.com/user-attachments/assets/b2155983-690f-4def-8827-751b5876298e)

One site I have uses the post ID as the post link. With the change in 6.6, it no longer shows the post ID, but instead shows the post slug. The permalink settings are ignored.

Another site prefixes the post ID ahead of the slug (123-post-title) and the link shown is only the post title (post-title).

## What is your proposed solution?

The link on the sidebar should show the link with the permalink settings applied so that the user can see the full URL without an extra interaction.

**Comment by Mamaduka** [comment]
IIRC, part of it is by design, but there's also room for improvement.

Cc-ing @jasmussen and @jameskoster for design feedback.

**Comment by jasmussen** [comment]
This is valid, if tricky. As part of the ongoing effort to migrate this particular panel to a DataForm, I wonder if @ntsekouras or @oandregal have any thoughts?

One thing that stands out is the mention of a custom permalink structure. That doesn't necessarily mean the full URL must be shown, given space is at a premium here, I wonder if it would make sense to show everything that comes after the _domain_, e.g. `2026/06/scandinavian-tristesse`, as opposed to just `scandinavian-tristesse`? The flyout would need improvements too, a few are already tracked in #63700.

**Comment by ntsekouras** [comment]
That's a hard one. I'm wondering if the `link` label at the time of this issue creation (see the screenshots) added more gravity. The label is `slug` [for a while](https://github.com/WordPress/gutenberg/pull/63669) , without solving the request for seeing the full URL though without opening the dialog. 

That's a hard one. I'm wondering if the `Link` label at the time of this issue creation (see the issue's screenshot) added more gravity. The label is `Slug` for a while now (renamed in #63669) and it is accurate, without solving the request for seeing the full URL without opening the dialog though. 

Although for permalink structures without `%postname%` (like the post id one in the report), we still show a slug that isn't part of the URL at all. Maybe we should show the link in that case, like we do for the front page? 🤔

If we really wanted the URL, could we add a second line with it like we do in the dialog? It would be quite noisy though, so do we want to do it even if there wasn't much traction in this issue all these years?

>I wonder if it would make sense to show everything that comes after the domain, e.g. 2026/06/scandinavian-tristesse, as opposed to just scandinavian-tristesse?

That could create similar confusion maybe, since the label is `slug` and we'll be showing something else? 🤔

**Comment by jasmussen** [comment]
Just to be clearer about what I'm suggesting; instead of this:

<img width="282" height="533" alt="Image" src="https://github.com/user-attachments/assets/ddd544f6-9fba-4ef8-87e7-50a041df05a7" />

We could do this:

<img width="277" height="531" alt="Image" src="https://github.com/user-attachments/assets/4c92c7b3-9bbd-4ea0-b9e1-7d9e05ef75ae" />

It doesn't wrap particularly nicely, but that would always be somewhat forced, and we could add in a break-with-dash piece to it.

**Comment by ntsekouras** [comment]
>Just to be clearer about what I'm suggesting; instead of this:

Yes. My `similar confusion maybe` part above is about that because the label is `slug`, but (from your example) `2026/06/scandinavian-tristesse` isn't the actual slug. Would that be a problem?

**Comment by jasmussen** [comment]
The flyout would still show just the slug part as editable 🤔

**Comment by jameskoster** [comment]
It's possible I'm misunderstanding the nature of 'custom slug' in this context, but I don't think @davidlim is referring to permalink structure settings. This appears to be something coded into the entities slug. If so, shouldn't we just show the full slug? To use the example from the OP, the slug should appear as `2203-this-is-a-test`. Since the custom part of the slug is uneditable it would appear as part of the prefix in the input, like the slash.
